### PR TITLE
chore: add `FilterInPlace` method to maps and update module

### PR DESCRIPTION
**Pull request by DmitriyMV**
- run rekres
- Deprecate `maps.Clone` (implemented in runtime since 1.21), `maps.Clear` (implemented as builtin function since 1.21) and `maps.Copy` functions.
- Fix typos.
- Add `FilterInPlace` method to key-value containers.
- Add `Len` method to key-value containers.
- Use `clear` builtin in `ConcurrentMap.Clear`.
- Use `slices` package instead of `sort` for sorting in tests.

### CODE REVIEW

**Review by smira — APPROVED** [review]

**Review by frezbo — COMMENTED** [review]
we need a PR first to enable actions

**Comment by smira** [comment]
> we need a PR first to enable actions

This still runs Drone still, so we should be safe? Next PR I propose to remove `.drone.yml` and test it?

File: maps/maps.go [hunk 1]
```diff
@@ -5,8 +5,10 @@
 // Package maps contains the generic functions for maps.
 package maps
 
-// NOTE(DmitriyMV): I tried to implement this generic functions to be as perfomant as possible.
-// However, I couldn't find a way to do it, since Go (1.18 at the time of writing) cannot inline closures if (generic)
+import "maps"
+
+// NOTE(DmitriyMV): I tried to implement this generic functions to be as performant as possible.
+// However, I couldn't find any way to do it, since Go (1.18 at the time of writing) cannot inline closures if (generic)
```
Comment (frezbo) [inline maps/maps.go:11]: does this matter anymore? since we're already in 1.21?

**Review by frezbo — APPROVED** [review]

File: maps/maps.go (hunk 1 above)
Comment (DmitriyMV) [reply maps/maps.go:11]: Inliner didn't got any smarter. They plan to do something in 1.22, but for now I think we should keep this comment.

**Pull request merged**